### PR TITLE
Remove a bunch of redundancy in variables and header text styles

### PR DESCRIPTION
**Pull request by kylefiedler**
These changes are to  simplify bitters and are opinionated based on building web applications and not web sites.

* Remove header styles because they always get changed. The initial styles just add overhead to be changed. 
* Remove `blockquote` and `cite` styles. I've hardly ever used them and changed them when I have.
* Create `$base-spacing-small` for all the places where we were using `$base-spacing / 2`
* Remove unnecessary duplication in form variables.
* Remove unnecessary duplication in link and button variables.
* Remove `$base-background-color` as it rarely is used more than once.

### CODE REVIEW

File: app/assets/stylesheets/_forms.scss [hunk 1]
```diff
@@ -1,7 +1,7 @@
 fieldset {
   background: lighten($base-border-color, 10);
   border: $base-border;
-  margin: 0 0 ($base-spacing / 2) 0;
+  margin: 0 0 $base-spacing-small 0;
```
Comment (tysongach) [inline app/assets/stylesheets/_forms.scss]: The last `0` can be removed here…

**Comment by kylefiedler** [comment]
Updated

**Comment by tysongach** [comment]
:+1:

**Comment by tysongach** [comment]
The [Example Page](http://bitters.bourbon.io/example.html) and the screenshot of the Example Page on the website should be updated for reflect these changes once merged in and released.

**Comment by kylefiedler** [comment]
@tysongach how is this: https://github.com/thoughtbot/bitters/pull/144

**Comment by tysongach** [comment]
:tada: LGTM

File: app/assets/stylesheets/_variables.scss [hunk 2]
```diff
@@ -18,6 +12,7 @@ $header-line-height: 1.25;
 // Other Sizes
 $base-border-radius: 3px;
 $base-spacing: $base-line-height * 1em;
+$small-spacing: $base-spacing / 2;
```
Comment (tysongach) [inline app/assets/stylesheets/_variables.scss]: If we’re adding a `$small-spacing` variable, should we also add `$large-spacing` (with a value of `$base-spacing * 2`)?

File: app/assets/stylesheets/_variables.scss [hunk 3]
```diff
@@ -26,29 +21,14 @@ $dark-gray: #333;
 $medium-gray: #999;
 $light-gray: #DDD;
 
-// Background Color
-$base-background-color: white;
```
Comment (tysongach) [inline app/assets/stylesheets/_variables.scss]: I would like to suggest leaving this variable in. Will and I built a buttons component yesterday, it was three buttons side by side with a slight border between them. We wanted that border to match the background, as if it was knocked out. This variable was exactly what we needed.

File: app/assets/stylesheets/_variables.scss (hunk 2 above)
Comment (kylefiedler) [inline app/assets/stylesheets/_variables.scss]: I added small spacing because it is used frequently in Bitters. I don't think we have a need for `$large-spacing` so I prefer not to include it.

File: app/assets/stylesheets/_variables.scss (hunk 3 above)
Comment (kylefiedler) [inline app/assets/stylesheets/_variables.scss]: Have you used it any other time? Should that be just a variable for the button component? I removed this because I've used it so infrequently. I think in most cases it is making the code a little more complex unnecessarily.

Comment (tysongach) [inline app/assets/stylesheets/_variables.scss]: Alright. It can be added on an as-needed basis. I don't think it should be with the button; it's not button-specific.

File: app/assets/stylesheets/_variables.scss [hunk 4]
```diff
@@ -26,29 +21,14 @@ $dark-gray: #333;
...
 $base-border: 1px solid $base-border-color;
 
 // Forms
-$form-border-color: $base-border-color;
-$form-border-color-hover: darken($base-border-color, 10);
-$form-border-color-focus: $base-accent-color;
-$form-border-radius: $base-border-radius;
-$form-box-shadow: inset 0 1px 3px rgba(black,0.06);
+$form-box-shadow: inset 0 1px 3px rgba(black, 0.06);
 $form-box-shadow-focus: $form-box-shadow, 0 0 5px rgba(darken($form-border-color-focus, 5), 0.7);
```
Comment (tysongach) [inline app/assets/stylesheets/_variables.scss]: Undefined variable: `$form-border-color-focus`

File: app/assets/stylesheets/_variables.scss [hunk 5]
```diff
@@ -26,29 +21,14 @@ $dark-gray: #333;
...
 $base-border-color: $light-gray;
 $base-border: 1px solid $base-border-color;
 
 // Forms
-$form-border-color: $base-border-color;
-$form-border-color-hover: darken($base-border-color, 10);
-$form-border-color-focus: $base-accent-color;
-$form-border-radius: $base-border-radius;
-$form-box-shadow: inset 0 1px 3px rgba(black,0.06);
+$form-box-shadow: inset 0 1px 3px rgba(black, 0.06);
```
Comment (tysongach) [inline app/assets/stylesheets/_variables.scss]: `black` » `#000`

File: app/assets/stylesheets/_variables.scss [hunk 6]
```diff
@@ -26,29 +21,14 @@ $dark-gray: #333;
...
-// Background Color
-$base-background-color: white;
-
 // Font Colors
 $base-font-color: $dark-gray;
-$base-accent-color: $blue;
-
-// Link Colors
-$base-link-color: $base-accent-color;
-$hover-link-color: darken($base-accent-color, 15);
```
Comment (tysongach) [inline app/assets/stylesheets/_variables.scss]: `$hover-link-color` is being called on Line 38 in `_typography.scss`

File: app/assets/stylesheets/_variables.scss (hunk 5 above)
Comment (whmii) [inline app/assets/stylesheets/_variables.scss]: awe :( I like named colors. I know it goes against the [styleguide](https://github.com/thoughtbot/guides/blob/master/style/sass/README.md#formatting) but i'll use `whitesmoke` till i die. :kissing_heart:

File: app/assets/stylesheets/_variables.scss [hunk 7]
```diff
@@ -26,29 +21,14 @@ $dark-gray: #333;
...
-$base-button-color: $base-link-color;
-$hover-button-color: $hover-link-color;
+$action-color: $blue;
```
Comment (whmii) [inline app/assets/stylesheets/_variables.scss]: :+1:  happy to see this. much more clear

File: app/assets/stylesheets/_variables.scss (hunk 6 above)
Comment (whmii) [inline app/assets/stylesheets/_variables.scss]: I sure think we should keep these in. It's important to have a consistent relationship between the action color and its alt/hover state. Perhaps we should add `$base-action-color` and `$hover-action-color`. I feel silly when i'm styling buttons and writing `&:hover { background-color: shade($action-color, some-arbitrary-value)}`

Comment (kylefiedler) [inline app/assets/stylesheets/_variables.scss]: @whmii there isn't always a relationship between the hover button background color and the link hover color and those are the only two use cases I have for that variable.

File: app/assets/stylesheets/_variables.scss (hunk 7 above)
Comment (Magnus-G) [inline app/assets/stylesheets/_variables.scss]: This might be off-topic but what is the reason for `$blue` and the other specified colors? I always feel like those are either too specific or one variable-abstraction step too many.

I'd rather strike out allt the `// Colors` and just write `$action-color: #477dca;`

File: app/assets/stylesheets/_variables.scss [hunk 8]
```diff
@@ -18,37 +12,23 @@ $header-line-height: 1.25;
...
 
 // Colors
-$blue: #477DCA;
+$blue: #477dca;
 $dark-gray: #333;
 $medium-gray: #999;
-$light-gray: #DDD;
-
-// Background Color
-$base-background-color: white;
```
Comment (Magnus-G) [inline app/assets/stylesheets/_variables.scss]: Why remove this one? It's handy especially if you want to create an inset area

Comment (tysongach) [inline app/assets/stylesheets/_variables.scss]: I made a [comment](https://github.com/thoughtbot/bitters/pull/141#discussion_r23235079) about this, too (it’s collapsed). This is a handy variable for dealing with things like inset area.

Comment (kylefiedler) [inline app/assets/stylesheets/_variables.scss]: My point was if / when it's handy there isn't anything stoping you from putting it in. How often have you used an inset on a project?

File: app/assets/stylesheets/_variables.scss (hunk 7 above)
Comment (kylefiedler) [inline app/assets/stylesheets/_variables.scss]: The thought here was to be clear about what color `$action-color` was. Off the top of my head I don't have a clear sense of what `#477dca` is. I could totally be convinced of deleting more things though.

File: app/assets/stylesheets/_variables.scss (hunk 8 above)
Comment (Magnus-G) [inline app/assets/stylesheets/_variables.scss]: It's not just insets. I use it for plenty of things such as inverted texts and subtle shadings and inverted areas. I vote for keeping it.

File: app/assets/stylesheets/_variables.scss (hunk 7 above)
Comment (Magnus-G) [inline app/assets/stylesheets/_variables.scss]: I've started experimenting with global variables in Refills, taking as much as possible from Bitters. These are the names I've played around with instead of the more specific ones used here:

```
// Colors
$dark-color: #333;
$light-color: #ddd;
```

I've also found myself using both `base-font-color` and something such as `base-font-color-inverted` quite often, for dark areas (if `base-font-color` is dark, that is). Therefore I'm experimenting with the following for Refills:

// Font Colors
$base-font-color-dark: $dark-color;
$base-font-color-light: #fff;
$action-color-dark: #477dca;
$action-color-light: lighten($action-color-dark, 25%);

Ideally the variables for Refills and Bitters would be the same but they don't have to be. IF Refills gets variables...

Comment (kylefiedler) [inline app/assets/stylesheets/_variables.scss]: I think changing the colors should go in to another PR. @Magnus-G if you feel strongly about this naming please do it and we can have a discussion there.

Comment (Magnus-G) [inline app/assets/stylesheets/_variables.scss]: Excellent! I'll submit a PR as soon as this one is merged.
